Sources/SwifterSwift/SwiftStdlib/ArrayExtensions.swift

```diff
@@ -110,6 +127,17 @@ public extension Array where Element: Equatable {
         }
     }
 
+    /// SwifterSwift: Remove all duplicate elements removed using KeyPath to compare.
+    ///
+    /// - Parameter path: Key path to compare, the value must be Hashable.
+    /// - Returns: an array of unique elements.
```

Comment:
Not sure if this description is correct, I mean ... the elements itself are not unique but the value of the specified keyPath within the element. What you think?

Reply:
"an array of elements, whose value by KeyPayh is unique." something like this?

Reply:
Humm... maybe "an array of elements, whose values within the elements given key path are unique"?

Reply:
Actually this function shouldn't be returning anything since it mutates `self`.
Also, it looks like these functions exist in `ArrayExtensions.swift` and we should be removing them from there, if we're implementing them here.

Reply:
Removed returning values from functions, changed comments. 
"Also, it looks like these functions exist in ArrayExtensions.swift and we should be removing them from there, if we're implementing them here." I did not unsterstand it clearly, @guykogus , can explain it please?

Reply:
These functions have been implemented already in `ArrayExtensions.swift`. Since `Array` conforms to `Sequence` you can delete the implementations from that file, since it's a duplication.

Reply:
Ah, ok, did it

Reply:
> Actually this function shouldn't be returning anything since it mutates `self`.

Yeah, but this seems like a pattern here. I really not sure why, maybe @omaralbeik knows
But we have a lot of mutating ones that return self in Collection, Sequence and Array exts.
We could open an issue so someone could audit thoses and remove it, but it would be a breaking change in those API's. 
Let me know what you think  @omaralbeik @guykogus

Reply:
I'm waiting for discussion, let me know which version of function we should left.

Reply:
Damn, it would be a breaking change. It would be more correct for the mutating functions to not return `self`, I'm also curious to know why they do that.
In general there should be 2 forms for these kinds of functions: `mutating func remove...()` with no return value, and `func removing...() -> Self`. Honestly I wouldn't mind seeing these changes made now, but I guess we'll need to save it for when we go up to version 6.0 - i.e. when Swift 6.0 is released 😳

Reply:
So should I return self in mutating functions here, as in other functions from this extension?
I looked through code, there are something about 12 functions, that don't return self. I think that code in extension should be consistent and if other mutating functions return "Self", new functions should also return self, and due to this kind of fix will be breaking change it should be done in major release.